components/kserve/resources/servicemesh/routing/istio-kserve-local-gateway.tmpl.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: kserve-local-gateway
+  namespace: {{ .ControlPlane.Namespace }}
+spec:
+  selector:
+    knative: ingressgateway
+  servers:
+    - hosts:
+        - 'default.host'
+      port:
+        name: https
+        number: 8445
+        protocol: HTTPS
+      tls:
+        mode: ISTIO_MUTUAL
```

Comment:
Shouldn't it be SIMPLE?
Although, I'm not sure if this will be overridden by odh-model-controller.

Reply:
this file will be applied as part of the feature and owned by featuretraker. if the same resource will "Process found Delat" be by odh-model-controller https://github.com/opendatahub-io/odh-model-controller/blob/6ce6e868ee8c3aee01355026331357092d4f87a4/controllers/reconcilers/kserve_isvc_gateway_reconciler.go#L145, it wont get reconcile back to ISTIO_MUTUAL (value set here) . right?

Reply:
> Shouldn't it be SIMPLE?
Although, I'm not sure if this will be overridden by odh-model-controller.

This is just sample and never be overwritten by odh model controller. Why I add this part is because we can not create a gateway with this information.

>  it wont get reconcile back to ISTIO_MUTUAL (value set here) . right?
right. it won't.

Reply:
>I don't understand why you are adding a sample then

it failed to create the Gateway without the sample Server so I have to add it. If you know if there is a way to create the Gateway without any Servers, please let me know.

Reply:
Can't it be `servers: {}` so that the sample is not needed? or would it also be rejected?

Reply:
No, webhook does not allow it
```
DataScienceCluster resource reconciled with component errors: 1 error occurred:
* failed to create object istio-system/kserve-local-gateway: admission webhook "rev.validation.istio.io" denied the request: configuration is invalid: gateway must have at least one server
```

Reply:
Then, either we go with the sample, or we create the Gateway in odh-model-controller...

I'm not really sure what's better. I see `Gateways` more as platform setup...

Reply:
I think my main reason for confusion was this statement:

> This is just sample and never be overwritten by odh model controller. Why I add this part is because we can not create a gateway with this information.

IIUC this gateway will have more hosts added by knative to make it possible to access VS by non-mesh members. @Jooho Can you confirm if my understanding is correct?

Reply:
I confirm on behalf of Jooho with the subtle clarification that the hosts are going to be added by odh-model-controller, and it is not just `hosts` but whole `server` entries, so that each entry can  have its own certificate.

Thus, indeed, the sample may never be touched if we leave it there.

Reply:
I believe some doc/explanation won't harm then. Someone looking at it in 6m timeframe might have the same questions as I did :)